### PR TITLE
EP-2194 - EP upgrade

### PR DESCRIPTION
**Pull request by wrandall22**
[Jira](https://jira.cru.org/browse/EP-2194)

This PR merges all of the EP upgrade changes into master so we can go live with EP 8.1. All of this code was initially reviewed as it went into the `ep-upgrade` branch.

### CODE REVIEW

**Comment by wrandall22** [comment]
Not ready for go-live yet, but figured I'd get a PR out there.

**Comment by wrandall22** [comment]
Plan is to go live next Tuesday.

File: src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js
Comment (dr-bizz) [inline src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js:1]: We need to ensure that after merge with `master` we are using the new cruPayments.
```import * as cruPayments from '@cruglobal/cru-payments/dist/cru-payments'```

File: src/common/components/paymentMethods/bankAccountForm/bankAccountForm.component.js
Comment (dr-bizz) [inline src/common/components/paymentMethods/bankAccountForm/bankAccountForm.component.js:1]: Again with this page, we must ensure the new cruPayments is used after merge with `maaster`

File: src/app/analytics/analytics.factory.js
Comment (dr-bizz) [inline src/app/analytics/analytics.factory.js:1]: We'll need to merge this file carefully too.

File: .github/workflows/build.yml
Comment (dr-bizz) [inline .github/workflows/build.yml:1]: Do we still need these extra branches in GHA?

**Review by dr-bizz — CHANGES_REQUESTED** [review]

File: src/common/services/paymentHelpers/extractPaymentAttributes.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+function flattenObject (obj) {
```
[the remark this answers is not in the document]
Comment (wrandall22) [reply src/common/services/paymentHelpers/extractPaymentAttributes.js:1]: It is called in order service, profile service, and payment-methods component.

File: .github/workflows/build.yml
Comment (wrandall22) [reply .github/workflows/build.yml:1]: After we go live we won't need them, but I don't want to remove them until we do, so we can test changes quickly if needed. The current workflow is to put `ep-upgrade` into the stage/prod branches for deployment to their environments. It is unlikely we'll need to make front-end changes at this point, but if we do I'd have to put them back.

File: src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js
Comment (wrandall22) [reply src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js:1]: Yes, this is using the new version

File: src/common/components/paymentMethods/bankAccountForm/bankAccountForm.component.js
Comment (wrandall22) [reply src/common/components/paymentMethods/bankAccountForm/bankAccountForm.component.js:1]: Yep, we are. I've been pulling all the changes in from master as they happen.

File: src/app/analytics/analytics.factory.js
Comment (wrandall22) [reply src/app/analytics/analytics.factory.js:1]: This should have all of the changes from `master`. I did notice on line 156 we have `siteSubSection`, but it never seems to get a value.

Comment (wrandall22) [reply src/app/analytics/analytics.factory.js:1]: Looks like this variable was never initialized, unless, when it was a `var` it was set from elsewhere.

File: src/common/services/hateoasHelper.service.js [hunk 2]
```diff
@@ -18,9 +18,14 @@ class HateoasHelper {
...
-      ? linkObj.uri.replace(/^\//, '') // remove initial slash
-      : undefined
+    if (linkObj) {
+      return linkObj.uri
+        ? linkObj.uri.replace(/^\//, '') // remove initial slash
+        : linkObj.href
+          ? linkObj.href.replace(/^https?:\/{2}:?[0-9]{0,4}[a-z].*\/cortex(.*)/, '$1')
+          : undefined
+    }
+    return undefined
```
[the remark this answers is not in the document]
Comment (wrandall22) [reply src/common/services/hateoasHelper.service.js:28]: I think I'll leave this one as-is.

**Pull request merged**